modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/ModuleExtPlugin.java

```diff
@@ -0,0 +1,292 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.gradle.plugins;
+
+import com.liferay.gradle.plugins.extensions.LiferayExtension;
+import com.liferay.gradle.plugins.internal.util.GradleUtil;
+
+import groovy.lang.Closure;
+
+import java.io.File;
+import java.io.FileInputStream;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.jar.Attributes;
+import java.util.jar.Attributes.Name;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.java.archives.Manifest;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetOutput;
+import org.gradle.api.tasks.bundling.Jar;
+
+import org.osgi.framework.Version;
+
+/**
+ * @author David Truong
+ */
+public class ModuleExtPlugin implements Plugin<Project> {
+
+	public static final String MODULE_EXT_CONFIGURATION_NAME = "moduleExt";
+
+	public static final String UNZIP_MODULE_EXT_JAVA_TASK_NAME =
+		"unzipModuleExtJava";
+
+	public static final String UNZIP_MODULE_EXT_MANIFEST_TASK_NAME =
+		"unzipModuleExtManifest";
+
+	public static final String UNZIP_MODULE_EXT_RESOURCES_TASK_NAME =
+		"unzipModuleExtResources";
+
+	@Override
+	public void apply(final Project project) {
+		GradleUtil.applyPlugin(project, JavaPlugin.class);
+		GradleUtil.applyPlugin(project, LiferayBasePlugin.class);
+
+		Configuration moduleExtConfiguration = _addConfigurationModuleExt(
+			project);
+
+		_addTaskUnzipModulesExtJava(project, moduleExtConfiguration);
+
+		_addTaskUnzipModulesExtResources(project, moduleExtConfiguration);
+
+		Copy copy = _addTaskUnzipModulesExtManifest(
+			project, moduleExtConfiguration);
+
+		Jar jar = _configureJarTask(project, copy);
+
+		_configureDeployTask(project, jar);
+	}
+
+	private Configuration _addConfigurationModuleExt(final Project project) {
+		Configuration configuration = GradleUtil.addConfiguration(
+			project, MODULE_EXT_CONFIGURATION_NAME);
+
+		configuration.setDescription(
+			"Configures the module you would like to extend.");
+		configuration.setTransitive(true);
+		configuration.setVisible(false);
+
+		Configuration compileOnlyConfiguration = GradleUtil.getConfiguration(
```

Comment:
Can you please move this in a method called `_configureConfigurationCompileOnly`?